measure.go

```diff
@@ -2,49 +2,163 @@ package main
 	}
+
+	pktxmt := length
+	pktrcv := len(row)
+	var d stats.Float64Data = row
+	median, _ := d.Median()
+	max, _ := d.Max()
+	min, _ := d.Min()
+	mean, _ := d.Mean()
+	dev, _ := d.StandardDeviation()
+
+	if len(row) > 0 {
+		dataMap["rttmin"] = strconv.FormatFloat(min, 'f', 2, 64)
+		dataMap["rttmax"] = strconv.FormatFloat(max, 'f', 2, 64)
+		dataMap["rttavg"] = strconv.FormatFloat(mean, 'f', 2, 64)
+		dataMap["rttmdev"] = strconv.FormatFloat(dev, 'f', 2, 64)
+		dataMap["rttmedian"] = strconv.FormatFloat(median, 'f', 2, 64)
+	}
+	dataMap["pkttransmit"] = strconv.Itoa(pktxmt)
+	dataMap["pktreceive"] = strconv.Itoa(pktrcv)
+
+	return dataMap
+}
+
+type Tcpping struct {
+	Utility
+}
+
+func (u *Tcpping) MarshalJSONParamsToGraph(target model.NqmTarget, agent model.NqmAgent, row map[string]string) []ParamToAgent {
+	return new(Fping).MarshalJSONParamsToGraph(target, agent, row)
 }
 
-func fping() {
-	for {
-		func() {
-			probingCmd, targets, agentPtr, err := makeTasks()
-			if err != nil {
-				log.Println(err)
-				return
-			}
-			log.Println("Execution of the probing command:", probingCmd[0:9])
+func (u *Tcpping) ProbingCommand(targetAddressList []string) []string {
+	probingCmd := append([]string{"tcpping.sh"}, targetAddressList...)
+	return probingCmd
+}
+
+func (u *Tcpping) utilName() string {
+	return "tcpping"
+}
+
+func (u *Tcpping) CalcStats(row []float64, length int) map[string]string {
+	return new(Fping).CalcStats(row, length)
+}
+
+type Tcpconn struct {
+	Utility
+}
+
+func (u *Tcpconn) MarshalJSONParamsToGraph(target model.NqmTarget, agent model.NqmAgent, row map[string]string) []ParamToAgent {
+	var params []ParamToAgent
+	params = append(params, marshalJSONToGraph(target, agent, "tcpconn", row["time"]))
+	return params
+}
 
-			rawData := Probe(probingCmd)
-			jsonParams := MarshalIntoParameters(rawData, targets, agentPtr)
-			Push(jsonParams)
-		}()
+func (u *Tcpconn) ProbingCommand(targetAddressList []string) []string {
+	probingCmd := append([]string{"tcpconn.sh"}, targetAddressList...)
+	return probingCmd
+}
+
+func (u *Tcpconn) utilName() string {
+	return "tcpconn"
+}
 
-		dur := time.Second * time.Duration(GetGeneralConfig().Agent.FpingInterval)
-		time.Sleep(dur)
+func (u *Tcpconn) CalcStats(row []float64, length int) map[string]string {
+	dataMap := map[string]string{
+		"time": "-1",
+	}
+	if length != 1 {
+		log.Fatalln("[", u.utilName(), "] Error on Calculation of statistics")
+	}
+	if len(row) > 0 {
+		time := row[0]
+		dataMap["time"] = strconv.FormatFloat(time, 'f', 2, 64)
 	}
+	return dataMap
+}
+
+func measureByUtil(u Utility) {
+	probingCmd, err := Task(u)
+	if err != nil {
+		log.Println(err)
+		return
+	}
+	log.Println("[", u.utilName(), "] Measuring...")
+
+	rawData := Probe(probingCmd, u.utilName())
+	parsedData := Parse(rawData)
+	statsData := Calc(parsedData, u)
+	jsonParams := Marshal(statsData, u)
+	Push(jsonParams, u.utilName())
 }
 
 func Measure() {
-	go fping()
-	go tcpping()
-	go tcpconn()
+	go func() {
```

Comment:
It should be controlled by HBS not `cfg.json`, please refer to [open-falcon agent](https://github.com/Cepave/agent/blob/develop/cron/collector.go#L28).